src/git-source-provider.ts

```diff
@@ -0,0 +1,285 @@
+import * as core from '@actions/core';
+import * as coreCommand from '@actions/core/lib/command';
+import * as fs from 'fs';
+import * as fsHelper from './fs-helper';
+import * as gitCommandManager from './git-command-manager';
+import * as io from '@actions/io';
+import * as path from 'path';
+
+let authConfigKey = `http.https://github.com/.extraheader`;
+
+export async function getSource(
+    repositoryPath: string,
+    repositoryOwner: string,
+    repositoryName: string,
+    ref: string,
+    commit: string,
+    clean: boolean,
+    fetchDepth: number,
+    lfs: boolean,
+    accessToken: string) {
+
+    core.info(`Syncing repository: ${repositoryOwner}/${repositoryName}`);
+    let repositoryUrl = `https://github.com/${encodeURIComponent(repositoryOwner)}/${encodeURIComponent(repositoryName)}`;
+
+    // Remove conflicting file path
+    if (fsHelper.fileExistsSync(repositoryPath)) {
+        await io.rmRF(repositoryPath);
+    }
+
+    // Create directory
+    let isNew = false;
+    if (!fsHelper.directoryExistsSync(repositoryPath)) {
+        isNew = true;
+        await io.mkdirP(repositoryPath);
+    }
+
+    // Git command manager
+    core.info(`Working directory is '${repositoryPath}'`);
+    let git = await gitCommandManager.CreateCommandManager(repositoryPath, lfs);
+
+    // Existing directory
+    if (!isNew) {
+
+        let recreate = false;
+
+        // Fetch URL does not match
+        if (!fsHelper.directoryExistsSync(path.join(repositoryPath, '.git')) ||
+            repositoryUrl != await git.tryGetFetchUrl()) {
+
+            recreate = true;
+        }
+        // Fetch URL matches
+        else {
+            // Delete any index.lock and shallow.lock left by a previously canceled run or crashed git process
+            let lockPaths = [
+                path.join(repositoryPath, '.git', 'index.lock'),
+                path.join(repositoryPath, '.git', 'shallow.lock')
+            ];
+            for (let i = 0; i < lockPaths.length; i++) {
```

Comment:
would use for of here since you don't need the index variable and use const for the variable in the loop

Reply:
added eslint/prettier